### PR TITLE
Add Windows CI to the GitHub workflow

### PR DESCRIPTION
**Pull request by AMoo-Miki**
Signed-off-by: Miki <amoo_miki@yahoo.com>

### Description
Tests are expected to fail until #2963, #2964, #2965, and a future automation of ChromeDriver detection are merged in.
 
This should be manually backported to 1.x.
 
### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [X] Update [CHANGELOG.md](./../CHANGELOG.md)
- [X] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: .github/workflows/build_and_test_workflow.yml [hunk 1]
```diff
@@ -116,12 +171,64 @@ jobs:
...
+
+      - name: Configure git's autocrlf
+        run: |
+          git config --global core.autocrlf false
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup JDK
+        uses: actions/setup-java@v3
```
Comment (kavilla) [inline .github/workflows/build_and_test_workflow.yml:191]: how come?

File: .github/workflows/build_and_test_workflow.yml [hunk 2]
```diff
@@ -186,18 +349,18 @@ jobs:
         working-directory: ./artifacts
     strategy:
       matrix:
-        version: [ osd-2.0.0, osd-2.1.0, osd-2.2.0, osd-2.3.0, osd-2.4.0 ]
+        version: [ osd-2.0.0, osd-2.1.0, osd-2.2.0, osd-2.3.0 ]
```
Comment (kavilla) [inline .github/workflows/build_and_test_workflow.yml:352]: by accident?

**Review by kavilla — COMMENTED** [review]
do we just want to make matrices and just template this out i'd imagine the configure git autocrlf can be used by linux as well

File: .github/workflows/build_and_test_workflow.yml (hunk 2 above)
Comment (AMoo-Miki) [reply .github/workflows/build_and_test_workflow.yml:352]: yep; bad cherrypick. will fix in a moment.

File: .github/workflows/build_and_test_workflow.yml (hunk 1 above)
Comment (AMoo-Miki) [reply .github/workflows/build_and_test_workflow.yml:191]: I am using the bare Windows image which doesn't have JDK to run OS for the tests.

**Comment by ananzh** [comment]
some tests fail

**Comment by AMoo-Miki** [comment]
> some tests fail

That is expected. It is dependent on the other Windows related PRs to be in.

File: .github/workflows/build_and_test_workflow.yml [hunk 3]
```diff
@@ -25,18 +25,18 @@ env:
...
-    name: Build and Verify
+    name: Build and Verify on Linux
     steps:
       - name: Checkout code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
 
       - name: Setup Node
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v3
```
Comment (manasvinibs) [inline .github/workflows/build_and_test_workflow.yml:39]: Just curious to understand why are we bumping the version here?

Comment (AMoo-Miki) [reply .github/workflows/build_and_test_workflow.yml:39]: `v2` actions are all Node v12 actions and have been deprecated; `v3` actions are Node v16 based.

https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/

File: .github/workflows/build_and_test_workflow.yml [hunk 4]
```diff
@@ -25,18 +25,18 @@ env:
   OSD_SNAPSHOT_SKIP_VERIFY_CHECKSUM: true
 
 jobs:
-  build-lint-test:
+  build-lint-test-linux:
     runs-on: ubuntu-latest
     container:
       image: docker://opensearchstaging/ci-runner:ci-runner-rockylinux8-opensearch-dashboards-integtest-v2
```
Comment (manasvinibs) [inline .github/workflows/build_and_test_workflow.yml:31]: Should this image be also bumped to integtest-v3 based on the below version bump to v3 or is the same image?

Comment (AMoo-Miki) [reply .github/workflows/build_and_test_workflow.yml:31]: Nah. that is a custom image of ours. Those `v3`s are stuff that run on GH. However, we should figure out what blockers we have to using standard GH images.

**Review by manasvinibs — APPROVED** [review]

**Comment by AMoo-Miki** [comment]
> do we just want to make matrices and just template this out i'd imagine the configure git autocrlf can be used by linux as well

I think we should but the donno how to do conditional JDK install. Maybe we should do this when we start using standard linux images.

**Comment by joshuarrrr** [comment]
Do we have any benchmarks on the expected impact for this on the total duration of automated PR tests in general? While more testing is obviously better, I'm concerned that we'll be constantly waiting around for automation to complete (which I'm currently doing for a handful of backport PRs already...)

**Comment by AMoo-Miki** [comment]
> Do we have any benchmarks on the expected impact for this on the total duration of automated PR tests in general? 

You can see it here in action: https://github.com/AMoo-Miki/OpenSearch-Dashboards/pull/2

Bootstrapping is a substantially slower on Windows and it takes longer for Windows builds to end. It ends up being roughly 66 mins with this vs 56 without it.

**Review by joshuarrrr — APPROVED** [review]

**Review by ananzh — APPROVED** [review]

**Pull request merged**